cfpb/consumerfinance.gov
# Exempt some models from Wagtail's reference index

Wagtail 4.1 introduced [a new reference index](https://docs.wagtail.org/en/stable/releases/4.1.html#object-usage-reporting) that keeps track of Django model instance usage. In versions 4.1 and 4.2, this tracks instances from any Django model.

In 5.0 [this was changed](https://docs.wagtail.org/en/stable/releases/5.0.html#referenceindex-no-longer-tracks-models-used-outside-of-wagtail) so that only Wagtail-related models are tracked, meaning Pages, Images, Documents, Snippets, or modeladmin-registered models.

Since we're on version 4.1, we need to manually exempt our other Django models to prevent them from being included in the reference index. This reduces the size of the index and removes the overhead of tracking these objects when they are created, referenced, or deleted.

This commit exempts the following models:

- agreements.Agreement
- agreements.Issuer
- ask_cfpb.Answer
- ask_cfpb.Category
- ask_cfpb.NextStep
- data_research.County
- data_research.MetroArea
- data_research.MortgageBase (and derived classes)
- data_research.State
- login.PasswordHistoryItem
- paying_for_college.Alias
- paying_for_college.BAHRate
- paying_for_college.ConstantCap
- paying_for_college.ConstantRate
- paying_for_college.Contact
- paying_for_college.Nickname
- paying_for_college.Notification
- paying_for_college.Program
- paying_for_college.School
- prepaid_agreements.PrepaidAgreement
- prepaid_agreements.PrepaidProduct
- v1.CDNHistory

Once we're on Wagtail 5.0, we'll be able to revert this change.

## How to test this PR

To test, run `cfgov/manage.py rebuild_reference_index` to see that these models are exempted from the index (note that this may take a long time due to the index's current large size, https://github.com/wagtail/wagtail/pull/10703 attempts to fix this).

## Checklist
